services/cd-service/pkg/repository/transformer.go

```diff
@@ -3320,33 +3330,70 @@ func (c *envReleaseTrain) prognosis(ctx context.Context, state *State, transacti
 						CiLink:            teamLocks[lockId].CiLink,
 						SuggestedLifetime: teamLocks[lockId].SuggestedLifetime,
 					}
-					locksList = append(locksList, newLock)
-
+					teamLocksMap[lockId] = newLock
 				}
 
 				appsPrognoses[appName] = ReleaseTrainApplicationPrognosis{
 					SkipCause: &api.ReleaseTrainAppPrognosis_SkipCause{
 						SkipCause: api.ReleaseTrainAppSkipCause_TEAM_IS_LOCKED,
 					},
-					Locks:   locksList,
-					Version: 0,
-					Team:    teamName,
+					EnvLocks:  nil,
+					TeamLocks: teamLocksMap,
+					AppLocks:  nil,
+					Version:   0,
+					Team:      teamName,
 				}
 				continue
 			}
 		}
+		releases := c.AllLatestReleasesCache[appName]
+		var release uint64
+		if releases == nil {
+			release = 0
+		} else {
+			release = uint64(releases[len(releases)-1])
+		}
+		commitID, err := getCommitID(ctx, transaction, state, release, appName)
+		if err != nil {
+			logger.FromContext(ctx).Sugar().Warnf("could not write event data - continuing. %v", fmt.Errorf("getCommitIDFromReleaseDir %v", err))
+		}
+
+		existingDeployment, err := state.DBHandler.DBSelectLatestDeployment(ctx, transaction, appName, c.Env)
+		if err != nil {
+			return failedPrognosis(err)
+		}
+
+		var existingVersion *uint64 = nil
+		if existingDeployment != nil && existingDeployment.Version != nil {
+			var tmp2 = (uint64)(*existingDeployment.Version)
+			existingVersion = &tmp2
+		}
+		var oldReleaseCommitId = ""
+		if existingVersion != nil {
+			oldReleaseCommitId, _ = getCommitID(ctx, transaction, state, *existingVersion, appName)
+			// continue anyway, this is only for events
+		}
+
```

Comment:
This is the core of this PR.
We load a bunch of data here, now in the prognosis, not when applying.